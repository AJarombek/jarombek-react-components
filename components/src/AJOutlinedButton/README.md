### Overview

Reusable button component with a specific "outlined" style.  For a more generic button, see `AJButton`.

**Use cases:**

```javascript
import { AJOutlinedButton } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJOutlinedButton disabled={true}>Outlined Button</AJOutlinedButton>
```

### Files

| Filename                   | Description                                                                     |
|----------------------------|---------------------------------------------------------------------------------|
| `AJOutlinedButton.js`      | The reusable `AJOutlinedButton` component.                                      |
| `index.js`                 | Index file which exports the `AJOutlinedButton` component.                      |
| `styles.js`                | JSS styles for the `AJOutlinedButton` component.                                |
