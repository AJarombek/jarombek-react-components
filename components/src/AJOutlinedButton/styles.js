/**
 * Stylesheet for the AJOutlinedButton component using JSS.
 * Author: Andrew Jarombek
 * Date: 1/6/2020
 */

export default {
  ajOutlinedButton: {

  },
  ajOutlinedButtonEnabled: {

  },
  ajOutlinedButtonDisabled: {

  }
};
