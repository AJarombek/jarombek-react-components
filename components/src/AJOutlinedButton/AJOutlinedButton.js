/**
 * Reusable outlined button component.
 * @author Andrew Jarombek
 * @since 1/6/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';

import styles from './styles';
import AJBaseButton from '../AJBaseButton/AJBaseButton';

const useStyles = createUseStyles(styles);

/**
 * Component for a button that displays text.
 * @param children The contents of the text button.
 * @param onClick An action that occurs when the button is clicked.
 * @param disabled Whether the button is clickable or not.
 * @return {*} React elements representing a text button.
 */
const AJOutlinedButton = ({children, onClick, disabled=false}) => {
  const classes = useStyles();

  const enabledClasses = [
    'aj-outlined-button',
    classes.ajOutlinedButton,
    classes.ajOutlinedButtonEnabled
  ];

  const disabledClasses = [
    'aj-outlined-button',
    classes.ajOutlinedButton,
    classes.ajOutlinedButtonDisabled
  ];

  return (
    <AJBaseButton
      onClick={onClick}
      disabled={disabled}
      enabledClasses={enabledClasses}
      disabledClasses={disabledClasses}>

      {children}
    </AJBaseButton>
  );
};

AJOutlinedButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default AJOutlinedButton;
