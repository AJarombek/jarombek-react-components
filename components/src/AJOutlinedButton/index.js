import AJOutlinedButton from './AJOutlinedButton';

export default AJOutlinedButton;
