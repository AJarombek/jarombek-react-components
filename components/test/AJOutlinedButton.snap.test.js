/**
 * Snapshot test for the {@link ../../components/AJOutlinedButton} component.
 * @author Andrew Jarombek
 * @since 1/11/2020
 */

import React from 'react';
import AJOutlinedButton from '../src/AJOutlinedButton/AJOutlinedButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AJOutlinedButton children={<div>Snapshot Test</div>}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer.create(
    <AJOutlinedButton
      children={<div>Snapshot Test</div>}
      onClick={f=>f}
      disabled={true}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
