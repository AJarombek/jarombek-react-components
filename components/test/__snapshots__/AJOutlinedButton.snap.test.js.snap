// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-outlined-button ajOutlinedButton-0-2-1 ajOutlinedButtonEnabled-0-2-2"
>
  <button
    disabled={false}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;

exports[`renders correctly with all props 1`] = `
<div
  className="ajBaseButton-0-2-4 ajBaseButtonDisabled-0-2-6 aj-outlined-button ajOutlinedButton-0-2-1 ajOutlinedButtonDisabled-0-2-3"
  onClick={null}
>
  <button
    disabled={true}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;
